### DISCUSSION TITLE
yarn workspace and nextjs

### DISCUSSION THREAD

**Discussion by artecoop** [vercel/next.js#27628] [Help]
Hello, this problem is driving me crazy.

I have a yarn workspace with 3 workspaces:

- the server code, to publish to the cloud
- a client wrote in nextjs, that will be published on Vercel
- a common package which has all the shared types between server and client

all this code is written in TypeScript.

In the common folder, among various .ts files, I have a package.json file (maconi is the name of the project):

```
{
    "name": "@maconi/common",
    "version": "1.3.0"
}
```

Defining the workspace is this package.json file at the root folder.
```
{
    "description": "Maconi project",
    "author": "Valerio",
    "private": true,
    "workspaces": [
        "common",
        "server",
        "public"
    ]
}
```

The `public` workspace contains the nextjs website. Here if I need to import one type to a page (say category), I'm importing it like:
```
... 

import { Category } from '@maconi/common/category';

const Example: React.FC = () => {
    const categories = new Array<Category>();

    ...

    return (
        <>
        <div>...</div>
        </>
    );
};

export default Example;
```

Pushing everything to my github account, Vercel automatically pull new commits from it and build my website (the server code is using a similar idea in another cloud service).

Well, it worked like a boss, until today.

Reviewing the code, I added a new type and fixed some stupid casing in file names, renaming all file names lowercase.
Fixed all references on my code (`import { Product } from '@maconi/common/Product';` become `import { Product } from '@maconi/common/product';` and so on), I tried to build locally before pushing to github (and consequently triggering a Vercel build) with great success. 
But when building on Vercel now I receive this error:

```
15:35:13.023 | info  - Checking validity of types...
15:35:20.139 | Failed to compile.
15:35:20.139 | ./pages/categories.tsx:4:26
15:35:20.139 | Type error: Cannot find module '@maconi/common/category' or its corresponding type declarations.
15:35:20.139 |     2 | import { useEffect, useState, ReactNode, ReactElement, useContext, createContext } from 'react';
15:35:20.139 |     3 |
15:35:20.139 | > 4 | import { Category } from '@maconi/common/category';
15:35:20.139 |        |                                           ^
15:35:20.139 |     5  |
15:35:20.140 |     6  | type ContextProps = {
15:35:20.140 |     7  |     isLoading?: boolean;
15:35:20.164 | error Command failed with exit code 1.
15:35:20.165 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
15:35:20.184 | Error: Command "yarn run build" exited with 1
```

Apart some file name rename and an added type (not even in the Category.ts file), nothing changed.
I'm not even sure if this is a Next problem.

Server code is unaffected and pushing created a working deploy.

**Comment by PaulRBerg** [comment]
You have to enable [`externalDir`](https://github.com/vercel/next.js/discussions/26420) in `next.config.js`.
